tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerInvoker.MultiProbe.cs

```diff
@@ -0,0 +1,197 @@
+        /// <param name="states">Debugger state</param>
+        /// <returns>CallTarget return structure</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static DebuggerReturn EndMethod_StartMarker<TTarget>(TTarget instance, Exception exception, ref MethodDebuggerState[] states)
+        {
+            DebuggerReturn ret = default;
+
+            for (var i = 0; i < states.Length; i++)
+            {
+                ret = EndMethod_StartMarker(instance, exception, ref states[i]);
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// End Method with Return value invoker
+        /// </summary>
+        /// <typeparam name="TTarget">Target type</typeparam>
+        /// <typeparam name="TReturn">Return type</typeparam>
+        /// <param name="instance">Instance value</param>
+        /// <param name="returnValue">Return value</param>
+        /// <param name="exception">Exception value</param>
+        /// <param name="states">Debugger state</param>
+        /// <returns>LiveDebugger return structure</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static DebuggerReturn<TReturn> EndMethod_StartMarker<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, ref MethodDebuggerState[] states)
+        {
+            DebuggerReturn<TReturn> ret = default;
+
+            for (var i = 0; i < states.Length; i++)
+            {
+                ret = EndMethod_StartMarker(instance, returnValue, exception, ref states[i]);
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// End Method with Void return value invoker
+        /// </summary>
+        /// <param name="states">Debugger state</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void EndMethod_EndMarker(ref MethodDebuggerState[] states)
+        {
+            for (var i = 0; i < states.Length; i++)
+            {
+                ref var state = ref states[i];
+                EndMethod_EndMarker(ref state);
+            }
+        }
+
+        /// <summary>
+        /// Log exception
+        /// </summary>
+        /// <param name="exception">Exception instance</param>
+        /// <param name="states">Debugger state</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void LogException(Exception exception, ref MethodDebuggerState[] states)
+        {
+            try
+            {
+                for (var i = 0; i < states.Length; i++)
+                {
+                    ref var state = ref states[i];
+                    LogException(exception, ref state);
+                }
+            }
+            catch
+            {
+                // ignored
+            }
+        }
+
+        /// <summary>
+        /// Used to clean up resources. Called upon entering the (instrumentation's) finally block.
```

Comment:
```suggestion
        /// Used to clean up resources. Called upon entering the instrumented code's finally block.
```

Reply:
Done in e3443b9b41459db08459c5cbadf67dd5b0bff45c.